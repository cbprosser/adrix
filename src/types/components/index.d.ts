export * from './AdvancedSearchForm';
export * from './MTGStaticTypes';
export * from './MultipleCardDisplay';
export * from './SimpleCardSearchBar';
export * from './Symbol';
