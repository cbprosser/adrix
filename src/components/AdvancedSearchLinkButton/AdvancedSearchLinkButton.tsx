import { Button } from '@mui/material';
import { FC } from 'react';
import { Link } from 'react-router-dom';
import { sitemap } from '../../sitemap';

export const AdvancedSearchLinkButton: FC = () => (
  <Link to={sitemap.advancedSearch.urls[0]}>
    <Button
      sx={(theme) => ({
        marginTop: theme.spacing(1),
      })}
      variant="contained"
    >
      Advanced Search
    </Button>
  </Link>
);
