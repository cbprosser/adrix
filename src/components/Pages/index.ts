export * from './AdvancedSearch';
export * from './Home';
export * from './Results';
