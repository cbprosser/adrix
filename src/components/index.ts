export * from './AppHeader';
export * from './CardDisplay';
export * from './Pages';
export * from './SimpleCardSearchBar';
