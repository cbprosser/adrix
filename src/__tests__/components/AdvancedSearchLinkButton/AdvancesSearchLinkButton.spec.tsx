import { render } from '@testing-library/react';
import { AdvancedSearchLinkButton } from '../../../components/AdvancedSearchLinkButton';

jest.mock('react-router-dom', () => ({
  Link: (props: any) => (
    <div data-testingid="router-link-mock" {...props}>
      {props.children}
    </div>
  ),
}));

describe('AdvancedSearchLinkButton Suite', () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<AdvancedSearchLinkButton />);

    expect(asFragment()).toMatchSnapshot();
  });
});
