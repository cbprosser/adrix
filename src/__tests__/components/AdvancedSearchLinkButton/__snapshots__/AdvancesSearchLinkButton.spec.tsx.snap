// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedSearchLinkButton Suite Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    data-testingid="router-link-mock"
    to="/search/advanced"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-0-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Advanced Search
      <span
        class="MuiTouchRipple-root css-1-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
